httpie/utils.py

```diff
@@ -12,10 +12,29 @@
 RE_COOKIE_SPLIT = re.compile(r', (?=[^ ;]+=)')
 
 
+class JsonDictWithDuplicateKeys(OrderedDict):
+    """A specialized JSON dict with duplicate keys support.
+
+    """
+
+    def __init__(self, items):
+        # `json.dumps()` will check the underlying dict (`self` in our case) and outputs
+        # directly if it is empty. So we need to have something in the dictionary.
+        self[''] = ''
+        self._items = items
+
+    def items(self):
+        return self._items
+
+
 def load_json_preserve_order(s):
     return json.loads(s, object_pairs_hook=OrderedDict)
 
 
+def load_json_preserve_duplicate_keys(s):
```

Comment:
I hesitated to alter `load_json_preserve_order()` directly. LMK if you think it is better to do it.

Reply:
IMO it is good as-is. Let's not alter too much the behavior.

Reply:
Looking at where we use `load_json_preserve_order()` (with `=@` and `:=`), I think it makes sense to preserve the duplicate keys there as well. We never interpret the JSON body data; we only either send/save it somewhere or print it on the terminal. So we shouldn’t apply any cleanup (like the default duplicate key removal).